In [1]:
import pandas as pd
import statsmodels.api as sm
from statsmodels.formula.api import ols


In [2]:
# Creating a pandas dataframe for a One-Way ANOVA
data = {"species": ["cattle"]*16 + ["sheep"]*16 + ["goat"]*16,
        "T_gondii_positive": [87,32,25,94,106,13,114,5,87,32,112,7,107,12,65,54, 
                     87,32,29,42,63,8,57,14,55,16,68,3,64,7,37,34, 
                     47,24,10,53,59,4,59,4,52,11,60,3,54,9,38,25]}


df = pd.DataFrame(data)

# Fit the one-way ANOVA model using the ols method
model = ols("T_gondii_positive ~ C(species)", data=df).fit()
aov_table = sm.stats.anova_lm(model, typ=1)
print(aov_table)

              df        sum_sq      mean_sq         F    PR(>F)
C(species)   2.0   6610.666667  3305.333333  3.450724  0.040341
Residual    45.0  43104.000000   957.866667       NaN       NaN


The F-value is 3.450724 and the PR(>F) is 0.040341, which indicates that there is a significant difference in the mean of T. gondii positive results across the three species of cattle, sheep, and goat at (p < 0.05).